spec/integration/data_stream/fields/fields.spec.yml

```diff
@@ -416,6 +430,27 @@ spec:
           not:
             required:
               - metrics
+      - if:
+          properties:
+            type:
+              enum:
+                - boolean
+                - date
+                - double
+                - geo_point
+                - ip
+                - keyword
+                - long
```

Comment:
This list is not the complete list available.
According to [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/runtime-mapping-fields.html), the complete list is:
- boolean
- composite
- date
- double
- geo_point
- ip
- keyword
- long
- [lookup](https://www.elastic.co/guide/en/elasticsearch/reference/current/runtime-retrieving-fields.html#lookup-runtime-fields)

In this condition, just the [types that we support in the `type` field](https://github.com/elastic/package-spec/blob/main/spec/integration/data_stream/fields/fields.spec.yml#L42) are added. Missing types:
- composite
- lookup